OlympusDAO-Agent-Suit/BondContractControlVariableAdjustment/src/agent.spec.ts

```diff
@@ -0,0 +1,178 @@
+    callMock = jest.fn();
+    getStorageMock = jest.fn();
+    ethersMock = {
+      call: callMock,
+      getStorageAt: getStorageMock,
+      _isProvider: true, // Necessary for ethers accepting the mock as Provider
+    } as any;
+    handleTransaction = provideHandleTransaction(
+      REDEEM_HELPER_ADDRESS,
+      ethersMock
+    );
+  });
+
+  it("should return empty findings if not relevant event was emitted", async () => {
+    when(getStorageMock)
+      .calledWith(expect.anything(), expect.anything(), expect.anything())
+      .mockReturnValue(encodeParameter("uint256", 0));
+
+    const txEvent = new TestTransactionEvent().setBlock(1);
+
+    const findings = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should return a finding if relevant event is emitted from bond contract", async () => {
+    when(getStorageMock)
+      .calledWith(expect.anything(), expect.anything(), expect.anything())
+      .mockReturnValue(encodeParameter("uint256", 1));
+
+    BONDS_ADDRESSES.forEach((value, index) => {
+      when(callMock)
+        .calledWith(
+          {
+            data: redeemHelperInterface.encodeFunctionData("bonds", [index]),
+            to: REDEEM_HELPER_ADDRESS,
+          },
+          expect.anything()
+        )
+        .mockResolvedValue(encodeParameter("address", value));
+    });
+
+    const txEvent = new TestTransactionEvent()
+      .setBlock(2)
+      .addEventLog(
+        EVENT_ABI,
+        BONDS_ADDRESSES[0],
+        encodeParameters(
+          ["uint256", "uint256", "uint256", "bool"],
+          [12, 11, 10, true]
+        )
+      );
+
+    const findings = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([createFinding("12", "11", "10", "true")]);
+  });
+
+  it("should return empty findings if relevant event is emitted from no bond contracts", async () => {
+    when(getStorageMock)
+      .calledWith(expect.anything(), expect.anything(), expect.anything())
+      .mockReturnValue(encodeParameter("uint256", 2));
+
+    BONDS_ADDRESSES.forEach((value, index) => {
+      when(callMock)
+        .calledWith(
+          {
+            data: redeemHelperInterface.encodeFunctionData("bonds", [index]),
+            to: REDEEM_HELPER_ADDRESS,
+          },
+          expect.anything()
+        )
+        .mockResolvedValue(encodeParameter("address", value));
+    });
+
+    const txEvent = new TestTransactionEvent()
+      .setBlock(3)
+      .addEventLog(
+        EVENT_ABI,
+        BONDS_ADDRESSES[2],
+        encodeParameters(
+          ["uint256", "uint256", "uint256", "bool"],
+          [12, 11, 10, true]
+        )
+      )
+      .addEventLog(
+        EVENT_ABI,
+        createAddress("0x12"),
+        encodeParameters(
+          ["uint256", "uint256", "uint256", "bool"],
+          [12, 11, 10, true]
+        )
+      );
+
+    const findings = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
```

Comment:
should not there be 1 finding here? On the txn, one of the event log is directed to the bond contract right?

Reply:
No, because that contract wasn't found as a bond contract. The agent first will fetch the bonds length and after that will ask for the bond contract addresses, here I am saying that length is 2 and the contract emitting the event is the number 3 so it wasn't fetched as a bond contract.